tribe-ext-export-mec-events.php

```diff
@@ -0,0 +1,362 @@
+			), 'imports', 'tribe_aggregator_disable', false );
+
+			$setting_helper->add_field( 'mec_export_organizers', array(
+				'type' => 'html',
+				'html' => '<fieldset class="tribe-field tribe-field-html"><legend>' . esc_html__( 'Organizers', 'the-events-calendar' ) . '</legend><div class="tribe-field-wrap">' . $this->export_button( 'export_organizers', esc_html__( 'Export Organizers', 'the-events-calendar' ) ) . '<p class="tribe-field-indent description">' . esc_html__( 'Export your organizers from Modern Events Calendar.', 'the-events-calendar' ) . ' ' . '</p></div></fieldset><div class="clear"></div>',
+			), 'imports', 'tribe_aggregator_disable', false );
+
+			$setting_helper->add_field( 'mec_export_venues', array(
+				'type' => 'html',
+				'html' => '<fieldset class="tribe-field tribe-field-html"><legend>' . esc_html__( 'Venues', 'the-events-calendar' ) . '</legend><div class="tribe-field-wrap">' . $this->export_button( 'export_venues', esc_html__( 'Export Venues', 'the-events-calendar' ) ) . '<p class="tribe-field-indent description">' . esc_html__( 'Export your venues from Modern Events Calendar.', 'the-events-calendar' ) . '</p></div></fieldset><div class="clear"></div>',
+			), 'imports', 'tribe_aggregator_disable', false );
+		}
+
+		/**
+		 * Make a button to trigger the CSV creation process
+		 *
+		 * @param string $text
+		 *
+		 * @return string
+		 */
+		public function export_button( $type, $text = '' ) {
+			$text     = $text ? $text : __( 'Export Events', 'the-events-calendar' );
+			$settings = Tribe__Settings::instance();
+
+			// get the base settings page url
+			$url = apply_filters( 'tribe_settings_url', add_query_arg( array(
+				'post_type' => Tribe__Events__Main::POSTTYPE,
+				'page'      => $settings->adminSlug,
+				'tab'       => 'imports',
+			), admin_url( 'edit.php' ) ) );
+
+			$url = add_query_arg( array( $type => '1' ), $url );
+			$url = wp_nonce_url( $url, $type );
+
+			return sprintf( '<a href="%s" class="button">%s</a>', $url, $text );
+		}
+
+		/**
+		 * If the button is clicked, start working
+		 *
+		 */
+		public function listen_for_export_button() {
+			if ( empty( $_REQUEST['export_events'] ) && empty( $_REQUEST['export_organizers'] ) && empty( $_REQUEST['export_venues'] ) || ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_events' ) && ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_organizers' ) && ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_venues' ) ) {
+				return;
+			} elseif ( ! empty( $_REQUEST['export_events'] ) || wp_verify_nonce( $_REQUEST['_wpnonce'], 'export_events' ) ) {
```

Comment:
Hmm ... shouldn't we be using `&&` here instead of `||` (ie, _the request was to export_events **and** the nonce checks out)?_

Reply:
Good call! Thanks!